### PR TITLE
cpu: Rename CPU_ARCH to CPU_CORE

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
Currently cortexm and mips CPUs are using the variable `CPU_ARCH` to refer to the CPU core identifier. Even the documentation of this variable indicates that:
https://github.com/RIOT-OS/RIOT/blob/58cd1265178a767479850236f2b78598f0d217b2/makefiles/vars.inc.mk#L15

A naming discussion started in #14176 regarding our classification system for our CPUs. It makes sense to structure the names using the correct terms. This PR introduces `CPU_CORE` as a replacement for `CPU_ARCH`, and assigns the correct architecture value to cortexm and mips CPUs.

### Testing procedure
- Use `make dependency-debug` to check that the correct `CPU_CORE` and `CPU_ARCH` is assign to the boards

### Issues/PRs references
Somehow related to #14176

### CODE REVIEW

File: cpu/cortexm_common/Makefile.features [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
 
 # cortex-m4f and cortex-m7 provide FPU support
-ifneq (,$(filter $(CPU_ARCH),cortex-m4f cortex-m7))
+ifneq (,$(filter $(CPU_CORE),cortex-m4f cortex-m7))
   FEATURES_PROVIDED += cortexm_fpu
 endif
+
+# Set CPU_ARCH depending on the CPU_CORE
+ifeq ($(CPU_CORE),cortex-m0)
+  CPU_ARCH := armv6_m
```
[the remark this answers is not in the document]
Comment (leandrolanzieri) [reply cpu/cortexm_common/Makefile.features]: Yes, this makes sense. I went with the `armvxm` version. Also I updated the Kconfig for cortexm_common so it uses the same.

**Comment by fjmolinas** [comment]
I haven't taken a detailed look, but +1 for the general idea.

**Comment by kaspar030** [comment]
> This PR introduces `CPU_CORE` as a replacement for `CPU_ARCH`, and assigns the correct architecture value to cortexm and mips CPUs.

Could you sum up the state we have and will have? I'm honestly confused.

**Comment by leandrolanzieri** [comment]
> Could you sum up the state we have and will have? I'm honestly confused.

Currently `CPU_ARCH` variable is being used only in cortexm and mips CPUs to indicate the core of the CPU (instead of the architecture as the name suggests).

The PR moves these values to (the newly introduced variable) `CPU_CORE`, to use the proper terminology. `CPU_ARCH` is kept for the CPUs that already define it, but the correct architecture identifier is assigned to it.

For instance, a CPU that currently has `CPU_ARCH=cortex-m0plus`, with this PR will have `CPU_CORE=cortex-m0plus` and `CPU_ARCH=armv6m`.

**Comment by MrKevinWeiss** [comment]
@kaspar030 It makes sense to me, is there anything else that you can think of that this clarification of naming might effect (ie. anything with murdock or if something is used outside RIOT)?

**Comment by fjmolinas** [comment]
@leandrolanzieri this one needs a rebase, other than that can you uncrustify  `core/include/panic.h`?

**Comment by fjmolinas** [comment]
You can squash afterwards as well @leandrolanzieri!

**Comment by fjmolinas** [comment]
Added the testing lable, since @MrKevinWeiss tested!

**Comment by aabadie** [comment]
This needs a rebase now.

**Pull request merged**